js/plates-model/model-worker.js

```diff
@@ -23,12 +23,10 @@ function workerFunction () {
   if (props.playing) {
     if (config.snapshotInterval && model.stepIdx % config.snapshotInterval === 0) {
       if (model.stepIdx === 0) {
+        snapshots.length = 0
```

Comment:
I know I'm complicating snapshots a bit by using them for saving state during authoring. But, the two changes I made (clear snapshots at the beginning of a run, and never "double back" snapshots if you're at the beginning of a run) seemed reasonable enough, so I left it this way.

Reply:
I see. It took me a while to understand how it all works with authoring, but yeah, it makes sense. I'd just add a comment here that `snapshots.length = 0` is cleaning up authoring snapshots.